### ISSUE TITLE
Gamma issue when loading RAW images

### ISSUE THREAD

**Issue by fabiencastan** [OpenImageIO/oiio#2256]
### Problem

There is an inconsistency between the metatada `oiio:Colorspace` and the gamma applied on the image when loading a RAW image.

The problem has evolved over time with the different versions of OIIO:
- in previous versions, there was both a color shift and a gamma problem
- in version 1.8.12, everything was working fine and both problems were solved. The `oiio:Colorspace` returned when loading a raw file was `Linear` (and `raw:Colorspace` was `sRGB`).
- in 2.0.x, there is a gamma problem again. The `oiio:Colorspace` returned when loading a raw file is `sRGB` (no more `raw:Colorspace` metadata).

[Here](https://github.com/OpenImageIO/oiio/blob/master/src/raw.imageio/rawinput.cpp#L409) we set:
```
    // Set the gamma curve to Linear
    m_processor->imgdata.params.gamm[0] = 1.0;
    m_processor->imgdata.params.gamm[1] = 1.0;
```
And [few lines later](https://github.com/OpenImageIO/oiio/blob/master/src/raw.imageio/rawinput.cpp#L468) we set the colorspace:
```
m_processor->imgdata.params.output_color = c;
```
And the `oiio:Colorspace` is set to the same value (`c` is the index for libraw and `cs` is the corresponding colorspace in oiio as a string):
```
        m_spec.attribute("oiio:ColorSpace", cs);
```
But AFAIU, the `output_color` only controls the primaries and not the gamma curve that we imposed to linear in all cases by setting the parameter `params.gamm` to 1.
So I think that the `oiio:Colorspace` should be set to linear in all cases.

For instance in the release 1.8.16, the code was:
```
    // Set the gamma curve to Linear
    m_spec.attribute("oiio:ColorSpace","Linear"); <<<< Set to linear in all cases
    m_processor->imgdata.params.gamm[0] = 1.0;
    m_processor->imgdata.params.gamm[1] = 1.0;
```

### Versions

* OIIO branch/version: 1.8.12 was working fine, 2.0.X doesn't
* OS: tested on both Linux and Windows
* C++ compiler: gcc/VS

**Comment by lgritz** [comment]
#2016 looks like the culprit. If that's the right word...

I think now that I see that I should be setting the gamm values based on what color space you request? I believe that I was assuming that when you set sRGB for output_color, that was giving me true sRGB -- not just the color primaries but also the nonlinear response. I get it now, I also need to set gamm.

So do you believe that the best behavior is that we should always return a linear color response, and that raw:ColorSpace should just specify the primaries? Or do you think that if you ask for "raw:Colorspace" == "sRGB", it ought to give you a true sRGB with the response curve as well?

To avoid confusion, maybe instead of "raw:Colorspace", the controlling parameter ought to be "raw:output_color" to emphasize that we are setting that parameter to libraw's decoder, not requesting a colorspace in the way that we use it everyplace else, which encompasses both the primaries and response curve?

**Comment by lgritz** [comment]
I had a talk about this with our color scientist, and I think I have a plan for how to proceed. Stay tuned, I will post a PR as soon as I can.

**Comment by lgritz** [comment]
Proposed fix in #2260 

If you want linear response (rather than sRGB response), you need to open with the configuration hint "raw:ColorSpace" = "linear". (Other ImageInput plugins will ignore the request, it's safe to ask all the time.)

**Comment by fabiencastan** [comment]
That's perfect!
